src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/JCloudsProvider.java

```diff
@@ -216,14 +217,30 @@ protected String buildScriptToExecuteString(InstanceScript instanceScript) {
     }
 
     private RunScriptOptions buildScriptOptions(InstanceScript instanceScript) {
-        logger.info("Script options: userLogin=" + getVmUserLogin());
-        return Optional.ofNullable(instanceScript.getCredentials())
-                       .map(credentials -> RunScriptOptions.Builder.runAsRoot(false)
-                                                                   .overrideLoginCredentials(new LoginCredentials.Builder().user(credentials.getUsername())
-                                                                                                                           .password(credentials.getPassword())
-                                                                                                                           .authenticateSudo(false)
-                                                                                                                           .build()))
-                       .orElse(RunScriptOptions.Builder.overrideLoginUser(getVmUserLogin()));
+        return Optional.ofNullable(instanceScript.getCredentials()).map(credentials -> {
+            if (credentials.getPrivateKey() != null) {
+                String username = Optional.of(credentials.getUsername())
+                                          .filter(StringUtils::isNotEmpty)
+                                          .filter(StringUtils::isNotBlank)
+                                          .orElse(getVmUserLogin());
+                logger.info("Key based authentication is provided. Script is going to be executed with the given private key and username=" +
+                            username);
+                return RunScriptOptions.Builder.runAsRoot(false)
```

Comment:
I wonder what this actually does. I hope that runnning the ProActive nodes without superuser permissions will not break some obscur workflow's tasks.

Reply:
Hopefully, having the private key in the credentials have been added in this commit. Then all existing infrastructures should currently use other credentials, so they should not fall in this case. Do you see it as a sufficient assumption? Otherwise, the username that is retrieved here is also configurable in the application.properties file. You could put root here if needed.

Reply:
Ok so you mean that the default value is `true`? If yes I think we should do the same to avoid permission issues. I don't really have a use-case in mind that absolutely requires root permissions (sucks by design :-D) but I'm pretty sure some workflow does. I personnaly think that creating an issue and setting this flag to `true` is the way to go.
In the future we should check what this flag does in practice, if it actually runs the command as root (maybe using sudo?) and in that case the boolean may be added in `application.properties`. BTW I don't know if using directly `root` as the user will work.